docs: Migrate i18n guide pages and code to Standalone

wardbell opened the pull request:
E2E tests passed locally

**Added `readme.md` to the example code**

This explains how to build and run the example for both English and French.

**Archived `i18n-optional-manual-runtime-locale.md`**

I was asked to cut out a substantial swath of this page by reviewer "zip-fa". All that remained was a brief paragraph.  I removed the page from the navigation after copying the remaining text to `i18n-optional-import-global-variants.md`. The page is still reachable by 3rd party links.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

wardbell committed:
docs: Migrate i18n guide pages and code to Standalone

E2E tests passed locally

**Added `readme.md` to the example code**

This explains how to build and run the example for both English and French.

**Archived `i18n-optional-manual-runtime-locale.md`**

I was asked to cut out a substantial swath of this page by reviewer "zip-fa". All that remained was a brief paragraph.  I removed the page from the navigation after copying the remaining text to `i18n-optional-import-global-variants.md`. The page is still reachable by 3rd party links.

zip-fa commented on aio/content/guide/i18n-optional-import-global-variants.md:
```diff
@@ -30,4 +36,4 @@ The following example imports the global variants for French \(`fr`\).
 <!-- end links -->
 
-@reviewed 2022-02-28
+@reviewed 2022-08-30
```
Its 2023 now

zip-fa commented on aio/content/guide/i18n-optional-manual-runtime-locale.md:
```diff
@@ -29,4 +37,4 @@ The following example sets the value of `LOCALE_ID` to `fr` for French.
 <!-- end links -->
 
-@reviewed 2022-02-28
+@reviewed 2022-08-30
```
Same. 2023

zip-fa commented on aio/content/examples/i18n/doc-files/main.1.ts:
```diff
@@ -1,12 +1,16 @@
-  enableProdMode();
-}
+bootstrapApplication(AppComponent, {
+  providers: [
```
Why do you need to provide LOCALE_ID here? Default locale is configured in angular.json (i18n.sourceLocale). Angular do provide automatically LOCALE_ID based on built code. This provider is helpful when you are running dev server i dont know..

wardbell replied:
It was in the `doc-files/app.module` class constructor in the original. 
```
@NgModule({
  imports: [ BrowserModule ],
  declarations: [ AppComponent ],
  providers: [ { provide: LOCALE_ID, useValue: 'fr' } ],
  bootstrap: [ AppComponent ]
})
export class AppModule { }
```
The `app.component` seemed the only logical place to move it.

Here is what the previous `i18n-optional-manual-runtime-locale.md` said (when it referred to `app.module`.
```
To manually set the runtime locale of an application to one other than the automatic value, complete the following actions.

1.  Search for the Unicode locale ID in the language-locale combination in the [`@angular/common/locales/`][UnpkgBrowseAngularCommonLocales] directory.
1.  Set the [`LOCALE_ID`][AioApiCoreLocaleId] token.

The following example sets the value of `LOCALE_ID` to `fr` for French.

<code-example header="src/app/app.module.ts" path="i18n/doc-files/app.module.ts" region="locale-id"></code-example>
```

Are you saying that I should eliminate all of this?

Perhaps that explains why the Stackblitz example and `ng serve` don't translate to FR. You have to run `ng server --configuration=fr`

BTW, do you know how I can make the Stackblitz translate? Can one tell Stackblitz to use `--configuration=fr`?

zip-fa replied:
Yeah i think you should eliminate provider from app.component.ts because it will confuse everyone (as you and me). I dont know how to force stackblitz to translate everything but i will try to do that tomorrow. Maybe it’s better to ask maintainers for help here? What is the real purpose behind providing LOCALE_ID by hand?

wardbell replied:
To repeat, what do you suggest I do about `LOCALE_ID`?  Remove everything from "To manually set .." through the code example?

zip-fa replied:
Yeah i suggest to remove LOCALE_ID and refer users to manual configuration in angular.json

wardbell replied:
Thanks. Will do.
And I figured out how to make Stackblitz work and added instruction for that to the examples `readme.md`

zip-fa replied:
How about adding example of using $localize fn inside component.ts? The guide has not covered this in example. Also searching by $localize gives no results. See: https://ng-dev-previews-fw--pr-angular-angular-51589-g89v2o7d.web.app/guide/i18n-example

zip-fa commented on aio/content/examples/i18n/src/main.ts:
```diff
@@ -1,7 +1,18 @@
-platformBrowserDynamic().bootstrapModule(AppModule)
-  .catch(err => console.error(err));
+bootstrapApplication(AppComponent, {
+  providers: [
```
Docs generated empty `providers` array:

```
, {
  providers: [
  ]
}
```

That would confuse newcomers.

See: https://ng-dev-previews-fw--pr-angular-angular-51589-g89v2o7d.web.app/guide/i18n-example

(main.ts tab)

MarkTechson replied:
@zip-fa I think the empty providers is fine? Am I missing something here?

zip-fa replied:
Yeah it’s totally fine but why do you need unused code which doesn’t affect anything for the end-user?

MarkTechson replied:
I was trying to understand your feedback. I agree that we can remove the application config here since there are no providers.

@wardbell - can you please remove the empty `AppConfig` from `bootstrapComponent`?

zip-fa replied:
Please see my previous comment about non-existing example for $localize function. I propose to add example usage of $localize inside this example. Also topic about “i18n and SSR” is not covered in docs, which is weird.

zip-fa replied:
Yeah. But $localize is widely supported and ok to use. Need to be documented well :)

wardbell replied:
Expanding the documentation to include `localize$` sounds great to me. 
But that's out-of-scope for the migration, right?
You can do that in a future revision, right?

zip-fa replied:
I never contributed to angular before to be honest. But yes, maybe we have to open an issue with “Add $localize to i18n example app” and impliment it in future. But my opinion that docs lacks of that kind of information and this kind of example should be covered too. How do you think would be better to do this? Make in next revisions on spend some time and cover now? Cant help you rn, i am on vacation :(

MarkTechson replied:
@zip-fa can you please open an issue and we can ensure that it makes it into a future revision. For this specific work, the focus is on standalone components.

wardbell replied:
I updated the display of `main.ts`. There is no longer a configuration object.

MarkTechson reviewed (approved):
Thanks for addressing the feedback, @wardbell! Other than the broken tests for the CI, this is good to go.

LGTM 🚀

wardbell commented:
> Thanks for addressing the feedback, @wardbell! Other than the broken tests for the CI, this is good to go.
> 
> LGTM 🚀

I believe that E2E tests pass now

AndrewKushnir commented:
This PR was merged into the repository by commit f6ccb163aaba790869e39512f0e73dac026379b6.
